pkg/runctl/controller.go

```diff
@@ -440,7 +440,7 @@ func (c *Controller) syncHandler(key string) error {
 			msg := run.GetMessage()
 			pipelineRun.UpdateMessage(msg)
 			pipelineRun.UpdateResult(result)
-			if err = c.changeState(pipelineRun, api.StateCleaning); err != nil {
+			if err = c.changeState(pipelineRun, api.StateCleaning, *run.GetCompletionTime()); err != nil {
```

Comment:
To be more consistent we should consider to also pass the same timestamp to UpdateResult one line above.

Reply:
Or add a function UpdateFinishedAt.

Reply:
ok, when we need it also on finished result so that the finishedAt is set also with that it makes sense to have a dedicated setter. Since we commit anyway after we called all relevant setters that is fine.

Reply:
Does not work well since we would have to "store" the timestamp somewhere in the pipeline run.